### DISCUSSION TITLE
Can't see line completion with solarized colors

### DISCUSSION THREAD

**Discussion by andrew222651** [xonsh/xonsh#4940] [Q&A] [answered]
I'm using Alacritty with a solarized dark color scheme. Using the solarized dark color scheme (or any other one I try) in xonsh, I can't see the autocomplete text that shows the suggestion for the rest of the line (i.e. what will be filled in if I press the Right key).
![2022-09-16-122236_434x158_scrot](https://user-images.githubusercontent.com/889657/190685648-6e0aae5f-fab0-4961-93cf-8ccf21758fe2.png)
Is there a way to fix this?

**Comment by gforsyth** [comment, marked as answer]
Hey @amacfie -- sorry about the styling issue, that's a bit odd.  
First thing, can you check the output of `xonfig` to make sure you are using the `prompt_toolkit` backend?  The `readline` backend doesn't have autosuggestions.

If that's not the issue, you can override any particular style element -- check the output of `xonfig colors` to see the token names and what they currently look like.

To set the auto-suggest text to a slightly aggressive shade of yellow, for instance, you can do:
```
__xonsh__.shell.shell.styler.override({"Token.PTK.AutoSuggestion": "#FFFF00"})
```

**Reply by andrew222651** [reply]
Thanks and yeah it's using `prompt_toolkit`. Anyway, the code you gave does make it visible -- however, if I put it in my `rc.xsh` it has an error, any fix?

**Reply by gforsyth** [reply]
Ahh, bit of a chicken-and-egg thing going on.  You can use the `on_post_init` event to delay evaluation of the style update until after the `__xonsh__` object has been populated.

```
@events.on_post_init
def update_style():
    __xonsh__.shell.shell.styler.override({"Token.PTK.AutoSuggestion": "#FFFF00"})
```

**Reply by andrew222651** [reply]
Awesome:)

**Reply by anki-code** [reply]
The PR with fix is welcome!

**Reply by andrew222651** [reply]
In xonsh 0.16.0 the above code gives an error: `AttributeError: 'NoneType' object has no attribute 'override'`. Any ideas?

**Reply by anki-code** [reply]
Please report the issue with stack trace.

**Reply by andrew222651** [reply]
The stack trace is just the line `__xonsh__.shell.shell.styler.override({"Token.PTK.AutoSuggestion": "#FFFF00"})` and that error

**Reply by anki-code** [reply]
What about switching $XONSH_SHOW_TRACEBACK to True before call?
